src/sql/workbench/services/connection/browser/passwordChangeDialog.ts

```diff
@@ -0,0 +1,162 @@
+
+
+	constructor(
+		@IThemeService themeService: IThemeService,
+		@IClipboardService clipboardService: IClipboardService,
+		@ILayoutService layoutService: ILayoutService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@ILogService logService: ILogService,
+		@IContextViewService private readonly contextViewService: IContextViewService,
+		@IConnectionDialogService private connectionDialogService: IConnectionDialogService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService
+	) {
+		super('', '', telemetryService, layoutService, clipboardService, themeService, logService, textResourcePropertiesService, contextKeyService, { dialogStyle: 'normal', hasTitleIcon: true });
+	}
+
+	public open(profile: IConnectionProfile, params: INewConnectionParams, uri: string) {
+		this._profile = profile;
+		this._params = params;
+		this._uri = uri;
+		this.render();
+		this.show();
+		this._okButton!.focus();
+	}
+
+	public override dispose(): void {
+
+	}
+
+	public override render() {
+		super.render();
+		this.title = DialogTitle;
+		this.titleIconClassName = TitleIconClass;
+		this._register(attachModalDialogStyler(this, this._themeService));
+		this._okButton = this.addFooterButton(OkText, () => this.handleOkButtonClick());
+		this._cancelButton = this.addFooterButton(CancelText, () => this.hide('cancel'), 'right', true);
+		this._register(attachButtonStyler(this._okButton, this._themeService));
+		this._register(attachButtonStyler(this._cancelButton, this._themeService));
+	}
+
+	protected renderBody(container: HTMLElement) {
+		const body = DOM.append(container, DOM.$('.change-password-dialog'));
+
+		this._loadingSpinner = new LoadingSpinner(container, { showText: true });
+		this._loadingSpinner.loadingMessage = passwordChangeLoadText;
+
+		const passwordRow = DOM.append(body, DOM.$('tr'));
+		DOM.append(passwordRow, DOM.$('td')).innerText = newPasswordText;
+		this._passwordValueText = new InputBox(DOM.append(passwordRow, DOM.$('.password-text')), this.contextViewService, {});
+		this._passwordValueText.inputElement.type = 'password';
+		this._register(attachInputBoxStyler(this._passwordValueText, this._themeService));
+
+		const confirmPasswordRow = DOM.append(body, DOM.$('tr'));
+		DOM.append(confirmPasswordRow, DOM.$('td')).innerText = confirmPasswordText;
+		this._confirmValueText = new InputBox(DOM.append(confirmPasswordRow, DOM.$('.confirm-text')), this.contextViewService, {});
+		this._confirmValueText.inputElement.type = 'password';
+		this._register(attachInputBoxStyler(this._confirmValueText, this._themeService));
+
+		const saveAndCloseCheckboxRow = DOM.append(body, DOM.$('tr'));
+		DOM.append(saveAndCloseCheckboxRow, DOM.$('td')).innerText = connectCheckboxText;
+		this._connectOnClose = new Checkbox(DOM.append(saveAndCloseCheckboxRow, DOM.$('.connect-check')), { label: connectCheckboxLabel });
```

Comment:
Do we need a checkbox for connect on close?  If user gets the error during connection shouldn't assume they want to connect (at least making the checkbox on by default).  Does SSMS have this checkbox in this scenario?

Reply:
I think this was added to allow users to not have to enter password again, once they reset and connect directly on close - but we can choose either way!

Reply:
This is the behavior with change password in AAD which is a good reference point too...

![image](https://user-images.githubusercontent.com/13396919/204669023-de58f974-a19f-4086-87af-a71ee980e3d6.png)

Once user sets a new password, they simply "Sign in" and are logged in.

Reply:
Personally I don't think we need the "Connect" checkbox (similar to SSMS).  Since the user already clicked Connect prior to getting the prompt we can infer their intent.  If the changed their mind or don't want to change password they can cancel the dialog.